WalletWasabi/WabiSabi/Client/OwnershipProofValidator.cs

```diff
@@ -0,0 +1,113 @@
+using NBitcoin;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using WalletWasabi.Blockchain.Transactions;
+using WalletWasabi.Crypto;
+using WalletWasabi.Stores;
+using WalletWasabi.Wallets;
+
+namespace WalletWasabi.WabiSabi.Client
+{
+	// OwnershipProofValidator validates the ownership proofs provided by the coordinator.
+	public class OwnershipProofValidator
+	{
+		public OwnershipProofValidator(IndexStore indexStore, TransactionStore transactionStore, IBlockProvider blockProvider)
+		{
+			IndexStore = indexStore;
+			TransactionStore = transactionStore;
+			BlockProvider = blockProvider;
+		}
+
+		public IndexStore IndexStore { get; }
+		public TransactionStore TransactionStore { get; }
+		public IBlockProvider BlockProvider { get; }
+
+		// Verifies the coins and ownership proofs. These proofs are provided by the coordinator who should
+		// have validate them before which means that when dealing with a honest coordinator all these proof
+		// are valid.
+		//
+		// In case one proof is invalid we have evidence enough to know the coordinator is malicious and we
+		// have to abort the process, notify the user and ban the coordinator.
+		public async ValueTask<int> VerifyOtherAlicesOwnershipProofsAsync(
+			uint256 roundId,
+			IEnumerable<(Coin Coin, OwnershipProof OwnershipProof)> othersCoins,
+			int stopAfter,
```

Comment:
Can you make a better naming on this? Can be longer.

Reply:
done.